src/plugins/discover/public/application/main/utils/resolve_data_view.ts

```diff
@@ -38,9 +38,6 @@ export function findDataViewById(
   dataViews: DataViewListItem[],
   id: string
 ): DataViewListItem | undefined {
-  if (!Array.isArray(dataViews) || !id) {
-    return;
-  }
   return dataViews.find((o) => o.id === id);
```

Comment:
I removed this code since its enforced by type safety. Let me know if there's some reason why this shouldn't be done.